test/bin/stress_testing.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+#
+# This script will enable stress scenarios to run MicroShift tests
+# with constricted resources
+
+set -uo pipefail
+
+SCRIPT_NAME=$(basename "$0")
+IP=$(nslookup "$(hostname)" | grep -v "#" | awk '/Address:/{print $2}')
+INTERFACE=$(ip route | grep default | awk -v ip="${IP}" '$0~ip{print $5}')
+
+WONDERSHAPER="/usr/local/bin/wondershaper"
+SPEEDTEST_CLI="/usr/local/bin/speedtest-cli"
+
+# bandwidth scenario
+function bandwidth_scenario {
+  bandwidth_scenario_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_bandwidth
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_bandwidth
+  elif [ "${ACTION}" == "get" ]; then
+    get_bandwidth
+  fi
+}
+
+function bandwidth_scenario_ready { 
+  # tc command
+  check_command "tc -V"
+
+  # wondershaper
+  if ( ! sudo ${WONDERSHAPER} -v  &> /dev/null);  then
+    sudo curl -Lo ${WONDERSHAPER} https://raw.githubusercontent.com/magnific0/wondershaper/master/wondershaper &> /dev/null
+    sudo chmod +x ${WONDERSHAPER} &> /dev/null
+    check_command "sudo ${WONDERSHAPER} -v"
+  fi
+
+  # speedtest-cli
+  if ( ! ${SPEEDTEST_CLI} --version &> /dev/null);  then
+    sudo curl -Lo ${SPEEDTEST_CLI} https://raw.githubusercontent.com/sivel/speedtest-cli/master/speedtest.py &> /dev/null
+    sudo chmod +x ${SPEEDTEST_CLI} &> /dev/null
+    check_command "${SPEEDTEST_CLI} --version"
+  fi
+}
+
+function enable_bandwidth {
+  echo -e "Bandwidth scenario enabled: max download and upload rate is ${VALUE} Kbps"
+  sudo ${WONDERSHAPER} -a "${INTERFACE}" -d "${VALUE}" -u "${VALUE}"
+}
+
+function disable_bandwidth {
+  echo -e "Bandwidth scenario disabled"
+  sudo ${WONDERSHAPER} -a "${INTERFACE}" -c || true
+}
+
+function get_bandwidth {
+  ${SPEEDTEST_CLI} --simple | tail -n2
+}
+
+# latency scenario
+function latency_scenario {
+  latency_scenario_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_latency
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_latency
+  elif [ "${ACTION}" == "get" ]; then
+    get_latency
+  fi
+}
+
+function latency_scenario_ready {
+  # tc command
+  check_command "tc -V"
+}
+
+function enable_latency {
+  echo -e "Latency scenario enabled: min latency is ${VALUE} ms"
+  sudo tc qdisc replace dev "${INTERFACE}" root netem delay "${VALUE}ms"
+}
+
+function disable_latency {
+  echo -e "Latency scenario disabled"
+  sudo tc qdisc delete dev "${INTERFACE}" root
+}
+
+function get_latency {
+  AVG_LATENCY=$(ping -c 3 8.8.8.8 | awk '/avg/{print $4}' | awk -F"/" '{printf "%d\n",$2}')
+  echo "${AVG_LATENCY}ms"
+}
+
+function tbd {
+  echo -e "This action is not implemented yet"
+}
+
+function check_command {
+  if ( ! $1 &> /dev/null ) ; then
+    echo -e "ERROR: $1: command not found"
+    exit 1
+  fi
+}
+
+# script info
+function version {
```

Comment:
I don't see the version function being used?

Reply:
yes, just removed it